src/single_random_draw.rs

```diff
@@ -0,0 +1,166 @@
+//! This library provides efficient algorithms to compose a set of unspent transaction outputs
+//! (UTXOs).
+
+#[cfg(any(test, feature = "rand"))]
+use crate::errors::LibError;
+
+use bitcoin::Amount;
+use bitcoin::blockdata::transaction;
+use bitcoin::FeeRate;
+use bitcoin::TxOut;
+use bitcoin::Weight;
+use crate::CHANGE_LOWER;
+use rand::seq::SliceRandom;
+
+/// Randomly select coins for the given target by shuffling the utxo pool and
+/// taking UTXOs until the given target is reached.
+#[cfg_attr(docsrs, doc(cfg(feature = "rand")))]
+pub fn select_coins_srd<R: rand::Rng + ?Sized>(
+    target: Amount,
+    outputs: &mut Vec<TxOut>,
+    fee_rate: FeeRate,
+    rng: &mut R)
+-> Result<Option<Vec<TxOut>>, LibError> {
+    outputs.shuffle(rng);
+
+    let mut selected_effective_value: Amount = Amount::ZERO;
+    let mut i = 0;
+    let mut result = Vec::new();
+
+    // The required size of a change output must be at least
+    // larger than the target by CHANGE_LOWER.  Therefore,
+    // CHANGE_LOWER + target becomes the new lower bound for the
+    // selection.
+    while selected_effective_value < target + CHANGE_LOWER && i < outputs.len() {
```

Comment:
nit: I'd find it more readable to use a for loop over the outputs and break within if target is reached, but might be a question of taste.

Reply:
I think since reading Edgar Dijkstra paper Gotos considered harmful, I try to avoid breaks or returns if possible.  Also Rust borrows from Ruby a lot with the implicit return at the end of functions, and it's popular to prefer that instead of writing return or break to a return.  Although, if a `for`  loop is more commonly used, and others find that easier to read, I'm fine with that instead.  In the end, I agree it's mostly a matter of taste as you say :)

Reply:
changed this to a for loop similar to the core implementation.